### ISSUE TITLE
Cabal-install install fails on unidentifiable dependency with no errors

### ISSUE THREAD

**Issue by Kludgy** [commercialhaskell/stack#4165]
### General summary/comments (optional)

Getting started with a new project on Windows 10, `lts-12.1`. Including dependency `gpipe` in either package.yaml or the cabal file causes `stack solver` cabal-install to fail with no visible errors.

### Steps to reproduce

1. `stack new emptyerrortest`
2. `cd emptyerrortest`
3. Add `gpipe` to `package.yaml` dependencies section.
4. `stack solver`

### Expected

`cabal-install` either succeeds, or fails with reasonable error messages.

### Actual

`cabal-install` fails with no visible errors:

```
Using configuration file: stack.yaml
Using cabal packages:
- .\

Using resolver: lts-12.1
Using compiler: ghc-8.4.3
Asking cabal to calculate a build plan...
Trying with packages from lts-12.1 as hard constraints...
cabal.EXE: Could not resolve dependencies:
trying: emptyerror-0.1.0.0 (user goal)
next goal: gpipe (dependency of emptyerror-0.1.0.0)
Dependency tree exhaustively searched.
Could not parse cabal-install errors:

>>>> Cabal errors begin
<<<< Cabal errors end

CallStack (from HasCallStack):
  error, called at src/Stack\Solver.hs:130:25 in stack-1.7.1-EmhVRHO1dvE2LPJS2dX4Zb:Stack.Solver
```

Verbose:
https://gist.github.com/Kludgy/84d350320052ac175309cb7a6263a61a

### Stack version

```
λ stack --version
Version 1.7.1, Git revision 681c800873816c022739ca7ed14755e85a579565 (5807 commits) x86_64 hpack-0.28.2
```

### Method of installation

Old official binary, downloaded from stackage.org or fpcomplete's package repository, followed by `stack upgrade`.

**Comment by Kludgy** [comment]
I suppose the correct solution here is identifying the 'no solution found' case somehow, if it is even possible.

**Comment by dbaynard** [comment]
@Kludgy Do indicate whether that solves your problem.

I suspect parsing the `cabal-install` errors is too much of a moving target; however it may be possible to modify the error displayed when there is such a `cabal-install` error. I've added the help wanted tag for this reason.

**Comment by Kludgy** [comment]
Right, thank you for the quick response. I'm oblivious to how much the cabal-install API helps its clients out at this point.

I did not get a prompt to add the correct version when I had mentioned the non-existent `gpipe` (lowercase) name.  Once I changed to `GPipe` of course, there was no issue.

It took a while to identify that this was a case sensitivity issue as I over-analysed why the extremely-highlighted cabal-install error section might be empty. :)  In my specific case a simple replacement of the normal error section with something like, `No further cabal-install errors were reported.`, or maybe even eliding that section altogether, would have clued me in.
